drivers/hmis/app/graphql/types/hmis_schema/has_events.rb

```diff
@@ -0,0 +1,41 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Types
+  module HmisSchema
+    module HasEvents
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def events_field(name = :events, description = nil, **override_options, &block)
+          default_field_options = { type: [Types::HmisSchema::Event], null: false, description: description }
+          field_options = default_field_options.merge(override_options)
+          field(name, **field_options) do
+            argument :sort_order, Types::HmisSchema::EventSortOption, required: false
+            instance_eval(&block) if block_given?
+          end
+        end
+      end
+
+      def resolve_events_with_loader(association_name = :events, **args)
+        load_ar_association(object, association_name, scope: apply_event_arguments(Hmis::Hud::Event, **args))
+      end
+
+      def resolve_events(scope = object.events, **args)
+        apply_event_arguments(scope, **args)
+      end
+
+      private
+
+      def apply_event_arguments(scope, user: current_user, sort_order: :date)
+        events_scope = scope.viewable_by(user)
```

Comment:
Hmm, I'm getting errors when I try to query for events/services/assessments now. Here is the query I tried and the error:

```
{
  enrollment(id: "46762") {
    id
    events(sortOrder: DATE) {
      nodes {
        id
      }
    }
    services(sortOrder: DATE) {
      nodes {
        id
      }
    }
    assessments(sortOrder: DATE) {
      nodes {
        id
      }
    }
  }
}

```


```
{
  "errors": [
    {
      "message": "undefined method `viewable_by' for :events:Symbol",
      "backtrace": [
        "/app/drivers/hmis/app/graphql/types/hmis_schema/has_events.rb:36:in `apply_event_arguments'",
        "/app/drivers/hmis/app/graphql/types/hmis_schema/has_events.rb:30:in `resolve_events'",
        "/app/drivers/hmis/app/graphql/types/hmis_schema/enrollment.rb:51:in `events'",
```